sopel/modules/wikipedia.py

```diff
@@ -272,18 +272,74 @@ def mw_section(server, query, section):
     return text
 
 
+def say_image_description(bot, trigger, server, image):
+    desc = mw_image_description(server, image)
+
+    if desc:
+        bot.say(desc, truncation=" […]")
+
+
+def mw_image_description(server, image):
+    """
+    Retrieves the description for the given image
+    """
+    params = "&".join([
+        "action=query",
+        "prop=imageinfo",
+        "format=json",
+        "indexpageids=1",
+        "iiprop=extmetadata",
+        "iiextmetadatafilter=ImageDescription",
+        "iilimit=1",
+        "titles={image}".format(image=image),
+    ])
+    url = "https://{server}/w/api.php?{params}".format(server=server, params=params)
+
+    response = get(url)
+
+    try:
+        query_data = response.json()["query"]
+        pageids = query_data["pageids"]
+        pages = query_data["pages"]
+
+        if pageids == [-1] or len(pageids) > 1:
+            # We either got no results, or multiple results, time to bail out
+            return None
+        else:
+            page = pages[pageids[0]]
+
+        raw_desc = page["imageinfo"][0]["extmetadata"]["ImageDescription"]["value"]
+
+        parser = WikiParser(image)
+        parser.feed(raw_desc)
+        desc = parser.get_result()
+        desc = ' '.join(desc.split())  # collapse multiple whitespace chars
+
+        return desc
```

Comment:
If these throw any exceptions, they won't be caught (these operations don't raise `LookupError`), so why put them inside the `try`?

Reply:
Good point, fixed in 9642880